### ISSUE TITLE
Make a new release?

### ISSUE THREAD

**Issue by miguelmarco** [gheber/kenzo#129]
I am planning to write an interface to Kenzo from Sage. As part or that effort, I would like to include Kenzo as an optional or experimental package inside Sage. It would be helpful for that to have a newer release (the only one you have here is from 2015). Can you please create a new one?

**Comment by gheber** [comment]
Do you mean another release tag under https://github.com/gheber/kenzo/releases ? (BTW, there are monthly Quicklisp releases.)

**Comment by miguelmarco** [comment]
Yes, that is what I mean. The idea is to have an url to download a tarball.

**Comment by miguelmarco** [comment]
Thank you Francis. Our plan is to include what you call Kenzo-P. More precisely, the code in this repo, since we need it to be easily loadable from embedded common lisp (it is the  lisp included in Sage, and there is already a pretty fast library interface between Sage and ecl).

At some point, we might need some auxiliary functions inside Kenzo to create simplicial sets from finite data (that is how Sage does it) in order to have a  more flexible interface. We could send you pull requests with these functions, or, if you prefear, we could mantain a fork with them (we would prefear to keep it all in just one fork though).

Also, I would greatly appreciate if you could add a clear license file, so there is no doubt that it is compatible with Sage license (which is GPL v3+)

**Comment by miguelmarco** [comment]
Also, I would like to ask you: is there an easy way to compile all the kenzo system in a single .fas file?

**Comment by sphyynx** [comment]
Francis Sergeraert => miguelmarco.
==================================================

/> Thank  you  Francis.    Our  plan  is  to  include   what  you  call
 > Kenzo-P. More precisely, the code in  this repo, since we need it to
 > be  easily  loadable from  embedded  common  lisp  (it is  the  lisp
 > included  in  Sage, and  there  is  already  a pretty  fast  library
 > interface between Sage and ecl).
/
   I guess Gerd has already studied the compatibility / ECL?

/> At some point,  we might need some auxiliary  functions inside Kenzo
 > to create simplicial  sets from finite data in order  to have a more
 > flexible  interface.  We  could send  you pull  requests with  these
 > functions, or, if you prefear, we could mantain a fork with them (we
 > would prefear to keep it all in just one fork though).
/
   Better  to  fork,  so  that  I can  guarantee  a  version  under  my
responsability.   Except   minor  changes   taking  account   of  some
incompatibility with  some or other  Lisp version, there won't  be any
more significant change in Kenzo.  I suppose your extra functions will
be useful, in which case it will be possible to merge them and give up
the fork. Gerd is more experienced than I, opinion?

/> Also, I  would greatly appreciate if  you could add a  clear license
 > file, so there  is no doubt that it is  compatible with Sage license
 > (which is GPL v3+)
/
   This was already asked for and I do not remember what have been then
done. I do not  think there is a real danger, so that  you may add the
license you like  in the forked version, which will  be finally merged
as well.

Bien cordialement.

**Comment by miguelmarco** [comment]
Thanks for the answers.  I agree it makes sense to fork so we can experiment with the changes we need, and maybe later merge.

**Comment by sphyynx** [comment]
Francis Sergeraert => miguelmarco.
==================================================

/> BTW, i did find the way to compile into a single .fas file.
 > (require :asdf)
 > (push #P\"./\" asdf:*central-registry*)
 > (require :kenzo)
 > (asdf:make-build :kenzo :type :fasl :monolithic t :move-here #P\"./\")
 > (quit)
/
   Good! The next time  I need the help of an asdl  wizard, I know what
the competent man is.

Bien cordialement.

**Comment by jcguu95** [comment]
What's the status of Kenzo-DVF and Sphynx? Any help needed?

**Comment by adlai** [comment]
> BTW, i did find the way to compile into a single .fas file.
> 
> ```
> ...
> (asdf:make-build :kenzo :type :fasl :monolithic t :move-here #P\"./\")
> ```
> 
> seems to do the trick. It produces a 4MB file, which is fine. I have just done trivial tests with it, but found no problem so far.

Alternatively, you could try wrapping critical sections with `eval-when` (at worst, each file in one huge block), and then concatenating all the source files. The advantage of this approach is that you'd have one file of source code, mirroring the monolithic compiled file; although honestly the source code is probably more accessible as separate files.
